src/app/directives/datatable/pagination.directive.ts

```diff
@@ -0,0 +1,139 @@
+import { Directive, Host, Input, OnInit } from "@angular/core";
+import { Direction, Filters, Sorting } from "@baw-api/baw-api.service";
+import { withUnsubscribe } from "@helpers/unsubscribe/unsubscribe";
+import { AbstractModel } from "@models/AbstractModel";
+import { DatatableComponent, TableColumn } from "@swimlane/ngx-datatable";
+import {
+  BehaviorSubject,
+  combineLatest,
+  map,
+  Observable,
+  switchMap,
+  takeUntil,
+  tap,
+} from "rxjs";
+
+type Page = number | null;
+type Sort<Model> = Sorting<keyof Model> | null;
+
+@Directive({
+  selector: "[bawDatatablePagination]",
+})
+export class DatatablePaginationDirective<Model extends AbstractModel>
+  extends withUnsubscribe()
+  implements OnInit
+{
+  @Input() public filters$: Observable<Filters<Model>>;
+  @Input() public models: (filters: Filters<Model>) => Observable<Model[]>;
```

Comment:
this name is sub-optimal. `models` implies a list of static things. What you're really asking here for is a function. Suggestions:

- `getModels`
- `modelsGetter`
- `fetchModelsFunction`